src/sdc11073/location.py

```diff
@@ -23,30 +42,32 @@ def __init__(self, fac=None, poc=None, bed=None, bld=None, flr=None, rm=None, ro
         self.rm = rm  # room
         self.bed = bed  # Bed
 
-    def mk_extension_string(self):
+    def mk_extension_string(self) -> str:
+        """Create a string usable in LocationContextStateContainer."""
         elements = self._get_extension_elements()
         values = [e[1] for e in elements]
         return '/'.join(values)
 
-    def _get_extension_elements(self):
-        """
-        :return: a list of (urlName, value) tuples
-        """
+    def _get_extension_elements(self) -> list[loc_param_value_tuple]:
+        """Return a list of (urlName, value) tuples ('bldng' instead of 'bld')."""
         identifiers = []
         for url_name, attr_name in self._url_member_mapping:
-            value = getattr(self, attr_name)
-            if value is None:
-                value = ''
+            value = getattr(self, attr_name) or ''
             identifiers.append((url_name, value))
         return identifiers
 
     @property
-    def scope_string(self):
+    def scope_string(self) -> str:
```

Comment:
why not
```python
@property
def scope(self) -> ScopeType:
  ...
```
convert scope to string should be handled in `ScopeType` and not here

Reply:
I think it is better located here because ScopesType -> SdcLocation and SdcLocation -> ScopesType conversions then are handled in the same place.

Reply:
ScopesType contains a list of strings. scopes_string creates only one the one that represents this location.